### PR TITLE
REVAI-3854: SuperApi features

### PR DESCRIPTION
**Pull request by kirillatrev**
REVAI-3854: Initial draft

### CODE REVIEW

File: src/main/java/ai/rev/speechtotext/ApiClient.java [hunk 1]
```diff
@@ -1,11 +1,7 @@
 package ai.rev.speechtotext;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/ApiClient.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/ApiInterface.java [hunk 2]
```diff
@@ -1,9 +1,6 @@
 package ai.rev.speechtotext;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/ApiInterface.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/NlpModel.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package ai.rev.speechtotext.models;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/NlpModel.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJob.java [hunk 4]
```diff
@@ -79,6 +79,12 @@ public class RevAiJob {
   @SerializedName("segments_to_transcribe")
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJob.java:79]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java [hunk 5]
```diff
@@ -136,6 +136,12 @@ public class RevAiJobOptions {
   @SerializedName("speaker_names")
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:136]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/Summarization.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/Summarization.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationJobStatus.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationJobStatus.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationOptions.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationOptions.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/Summary.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/Summary.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/TranslationJobStatus.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/TranslationJobStatus.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/Translation.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/Translation.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/asynchronous/TranslationLanguage.java [hunk 13]
```diff
@@ -0,0 +1,62 @@
+package ai.rev.speechtotext.models.asynchronous;
```
Comment (kirillatrev) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/TranslationLanguage.java:1]: Clean

File: src/main/java/ai/rev/speechtotext/models/NlpModel.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+import com.google.gson.annotations.SerializedName;
+
+/** Model type. */
```
Comment (dmtrrk) [inline src/main/java/ai/rev/speechtotext/models/NlpModel.java]: This looks like can be more verbose

File: src/main/java/ai/rev/speechtotext/models/NlpModel.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Returns the String value of the enumeration.
+     *
+     * @return The String value of the enumeration.
+     */
+    public String getModel() { return model; }
+
+    @Override
+    public String toString() {
+        return "{" + "model='" + model + '\'' + '}';
```
Comment (dmtrrk) [inline src/main/java/ai/rev/speechtotext/models/NlpModel.java:31]: what is the purpose of having it in this format? why can't consumer of this type format it as needed?

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java [hunk 16]
```diff
@@ -563,4 +569,30 @@ public List<SpeakerName> getSpeakerNames() {
...
+  /**
+   * Specifies summarization options.
+   *
+   * @param summarizationOptions Summarization options
+   */
+  public void setSummarizationOptions(SummarizationOptions summarizationOptions) {
+    this.summarizationOptions = summarizationOptions;
+  }
+
+  public TranslationOptions getTranslationOptions() {
```
Comment (dmtrrk) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:596]: I think here should be a comment

File: src/main/java/ai/rev/speechtotext/models/asynchronous/Summarization.java [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+ * A Summarization Job object provides all the information associated with a summarization job
+ * submitted by the user.
+ */
+public class Summarization {
+  /** User defined prompt. * */
+  @SerializedName("prompt")
+  private String prompt;
+
+  /** Standard or Premium AI backend. * */
+  @SerializedName("model")
```
Comment (dmtrrk) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/Summarization.java:20]: not sure we can have any other model type, but we can say something like `summarization model` and then `@see NlpModel`. then we can extend the NlpModel as we need without other changes

File: src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.google.gson.annotations.SerializedName;
+
+/** Summarization formatting options. * */
```
Comment (dmtrrk) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java]: is there a purpose for `* */`? I see it everywhere in this PR

File: src/main/java/ai/rev/speechtotext/models/NlpModel.java (hunk 15 above)
Comment (kirillatrev) [reply src/main/java/ai/rev/speechtotext/models/NlpModel.java:31]: I don't know what the intention was, but I follow the same pattern. JobOptions class at the top level calls .toString() on every member and expects this output.

Comment (dmtrrk) [reply src/main/java/ai/rev/speechtotext/models/NlpModel.java:31]: yeah, really. at last for some nested models we have this formatting.
@amikofalvy by any chance do you know what was the purpose?

File: src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java (hunk 18 above)
Comment (kirillatrev) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java]: This is how autoformatted puts it

Comment (dmtrrk) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java]: I did not find anything relevant in Google. Also, I don't see it like this in other files. Even in some of your files there is a single star only

Comment (kirillatrev) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/SummarizationFormattingOptions.java]: Okay. Thanks auto formatter! :-)
I'll spend some more time on this.

**Review by dmtrrk — APPROVED** [review]

**Pull request merged**